### PR TITLE
[#63] Remove usage of deprecated scikit-learn API in GridSearchCV

### PR DESCRIPTION
**Pull request by smurching**
Update `spark_sklearn.model_selection.GridSearchCV` to pass `fit_params` to `fit()` rather than as a constructor argument to its superclass (`sklearn.model_selection.GridSearchCV`).

Also added a comment explaining that the tests in test_grid_search_1.py are currently skipped. These tests attempt to run scikit-learn `GridSearchCV` tests against the spark-sklearn implementation. Most of the scikit-learn tests pass, but tests that purposefully raise Exceptions (e.g. [test_grid_search_failing_classifier](https://github.com/scikit-learn/scikit-learn/blob/d9fdd8b0d1053cb47af8e3823b7a05279dd72054/sklearn/model_selection/tests/test_search.py#L1280)) fail since PySpark wraps the exceptions in Py4JErrors. PR #62 is a WIP attempt to fix the scikit-learn tests.

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -65,7 +65,10 @@ More extensive documentation (generated with Sphinx) is available in the `python
 ## Changelog
 
 - 2015-12-10 First public release (0.1)
-- 2016-08-16 Minor release:
-   1. the official Spark target is Spark 0.2
+- 2016-08-16 Minor release (0.2.0):
+   1. the official Spark target is Spark 2.0
    2. support for keyed models
+- 2017-09-14 Minor release (0.2.2):
+   1. The official Spark target is Spark >= 2.1
```
Comment (phi-dbq) [inline python/README.md]: Do we want to update travis to also test Spark 2.2?

File: python/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 # This file should list any python package dependencies.
-scikit-learn==0.18.1
+scikit-learn>=0.18.1
```
Comment (phi-dbq) [inline python/requirements.txt]: Do we also want to impose a version upper bound?

File: python/spark_sklearn/grid_search.py [hunk 3]
```diff
@@ -124,12 +121,12 @@ class GridSearchCV(BaseSearchCV):
...
     GridSearchCV(cv=None, error_score=...,
            estimator=SVC(C=1.0, cache_size=..., class_weight=..., coef0=...,
-                         decision_function_shape=None, degree=..., gamma=...,
+                         decision_function_shape=..., degree=..., gamma=...,
                          kernel='rbf', max_iter=-1, probability=False,
                          random_state=None, shrinking=True, tol=...,
                          verbose=False),
-           fit_params={}, iid=..., n_jobs=1,
-           param_grid=..., pre_dispatch=..., refit=..., return_train_score=...,
+           fit_params=..., iid=..., n_jobs=1,
```
Comment (phi-dbq) [inline python/spark_sklearn/grid_search.py]: This is fine. Just wondering since the [original code in scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/0.18.1/sklearn/model_selection/_search.py#L791-L800) uses `fit_params={}` in the docstring, is our doctest somewhat different from theirs?

File: python/README.md [hunk 4]
```diff
@@ -65,7 +65,10 @@ More extensive documentation (generated with Sphinx) is available in the `python
 ## Changelog
```
Comment (phi-dbq) [inline python/README.md]: This is not directly related to this PR. 
But I wondering given that scikit-learn (and other python libraries) use `joblib.Parallel` heavily, was implementing a `joblib.Parallel` backend considered in the original design? If so, what's the reason that it was not taken? Thanks!

File: python/README.md (hunk 1 above)
Comment (smurching) [reply python/README.md]: Oops, yes we should! Thanks for the catch :)

I had added Spark 2.2 to the build in #60 but forgot to merge that commit back into the feature branch (spark-2.2-patch) for this PR. I guess since I didn't change .travis.yml in this PR Github didn't detect any conflicts etc.

File: python/requirements.txt (hunk 2 above)
Comment (smurching) [reply python/requirements.txt]: That's a good idea, I'll set an upper bound of 0.19 (the latest release) for now.

**Comment by phi-dbq** [comment]
Thanks, Sid! I will shift the review to @MrBago.

File: python/spark_sklearn/grid_search.py (hunk 3 above)
Comment (smurching) [reply python/spark_sklearn/grid_search.py]: Our doctest (as of your review) didn't actually test the spark-sklearn GridSearchCV implementation; it instead tested against scikit-learn's GridSearchCV implementation. I've since updated it to run against the spark-sklearn implementation.

**Comment by thunterdb** [comment]
@smurching thank you for catching this issue. Merging.
